### DISCUSSION TITLE
About including HTML files

### DISCUSSION THREAD

**Discussion by xplosionmind** [manunamz/jekyll-wikilinks#18] [Ideas]
As of right now, as pointed out by @manunamz in #9, only Markdown files present in the website are considered by the plugin. Nevertheless, there are a few use cases where scanning for wikilinks also in HTML files would be really useful.

- First and foremost, in the case I want to link `test.html` in a file named `example.md`; in `example.md` I would use `[[test]]` to link the file, but since it is an HTML file it is not seen by the plugin. This is a bummer
- The most intuitive use case, in which I want to link a Markdown or HTML file in an HTML file, by using double square brackets instead of the `a` tag.
- backlinks support: I would like to see backlinks from every page of the website, not only pages I wrote in markdown. As a workaround, currently I am using [a Liquid script of my creation](https://github.com/xplosionmind/tommi.space/blob/main/_includes/backlinks.html) which scans for the same URL in a page (only works in `PRODUCTION` environment)

I actually do not understand why HTML files are excluded… performance reasons?

I think supporting HTML files too would be useful and not at all a drawback for whom does not use HTML files linking

**Comment by manunamz** [comment]
So, after stumbling across the [gollum](https://github.com/gollum/gollum) project and reading through their wiki for a bit, I am starting to come around to this idea.  

I'm going to dig around the codebase and see what they did to handle [multiple formats](https://github.com/gollum/gollum/wiki/Formats-and-extensions). I am particularly keen to get something like `[[test.html]]` working, similarly to how image embeds work.

No promises with regard to scheduling though.
